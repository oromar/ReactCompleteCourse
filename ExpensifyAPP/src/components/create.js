import React from 'react'
import ExpenseForm from './expense-form'

export const CreateExpensePage = () => 
    <div>
        <h1>Add Expense</h1>    
        <ExpenseForm />
    </div>

