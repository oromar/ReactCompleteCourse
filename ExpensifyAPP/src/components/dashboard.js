import React from 'react'
import ExpenseList from './expense-list'
import ExpenseListFilters from './expense-list-filters'

export const ExpanseDashboardPage = () => (
    <div>
        <h1>Dashboard</h1>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)