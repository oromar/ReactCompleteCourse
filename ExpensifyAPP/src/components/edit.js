import React from 'react'
import {connect} from 'react-redux'
import ExpenseForm from './expense-form'


const EditExpensePage = (props) => {
    return (
    <div>
        <h1>Edit Expense</h1>
        <ExpenseForm 
            expense={props.expense}
            buttonLabel={"Edit Expense"}
            onSubmit={(expense) => {
                console.log('updated')
            }}

        />
    </div>
)}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((e) => e.id === props.match.params.id)
    }
}

export default connect(mapStateToProps)(EditExpensePage)