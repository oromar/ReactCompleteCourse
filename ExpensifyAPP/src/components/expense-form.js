import React from 'react'

export default class ExpenseForm extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            description: '',
            amount: 0,
            createdAt: 0,
            note: ''
        }

        this.onDescriptionChange = this.onDescriptionChange.bind(this)
        this.onAmountChange = this.onAmountChange.bind(this)
        this.onNoteChange= this.onNoteChange.bind(this)
    }

    onDescriptionChange(e) {
        const description = e.target.value
        this.setState(({description}))
    }   

    onAmountChange(e) {
        const amount = e.target.value
        this.setState(({amount}))
    }

    onNoteChange(e) {
        const note = e.target.value
        this.setState(({note}))
    }

    render() {
        return (
                <div>
                    <form>
                        <div>
                            <input 
                                type="text" 
                                placeholder="Description" 
                                autoFocus
                                value={this.state.description}
                                onChange={this.onDescriptionChange}
                            />
                        </div>
                        <div>
                            <input 
                                type="number" 
                                placeholder="Amount"
                                value={this.state.amount}
                                onChange={this.onAmountChange}
                                
                            />        
                        </div>
                        <div>
                           <textarea 
                                placeholder="Add note for your expense (optional)"
                                onChange={this.onNoteChange}
                                value={this.state.note}
                           >
                           </textarea>
                        </div>
                        <button>Add Expense</button>
                    </form>
                </div>
        )
    }
}