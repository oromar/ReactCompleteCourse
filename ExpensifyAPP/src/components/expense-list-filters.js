import React from 'react'
import {connect} from 'react-redux'
import {setTextFilter, setSortBy, setStartDate, setEndDate} from '../actions/filters'
import {DateRangePicker} from 'react-dates'

export class ExpenseListFilters extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            focusedInput:null,
        }
        this.onDatesChange = this.onDatesChange.bind(this)
        this.onTextChange = this.onTextChange.bind(this)
        this.onSortChange = this.onSortChange.bind(this)
        this.onFocusChange = this.onFocusChange.bind(this)
    }
    onDatesChange({startDate, endDate}) {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onTextChange(e) {
        this.props.onTextChange(e.target.value)
    }
    onSortChange(e) {
        this.props.onSortChange(e.target.value)
    }
    onFocusChange(focusedInput) {
        this.setState({focusedInput}) 
    }
    render() {
        return (
            <div>
                <input 
                    id="desc" 
                    onChange={this.onTextChange} 
                    type="text" value={this.props.filters.text}
                />        
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}                    
                    onFocusChange={this.onFocusChange}
                    startDateId='start'
                    endDateId='end'
                    focusedInput={this.state.focusedInput}
                    isOutsideRange={() => false}
                    numberOfMonths={1}
                    showClearDates={true}
                />
                <select 
                    id="sort" 
                    onChange={this.onSortChange}
                    value={this.props.filters.sortBy}>
                        <option value='date'>Date</option>
                        <option value='amount'>Amount</option>
                </select>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps =(dispatch) => ({
    onTextChange: (text) => dispatch(setTextFilter(text)),
    onSortChange: (sortBy) => dispatch(setSortBy(sortBy)),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)