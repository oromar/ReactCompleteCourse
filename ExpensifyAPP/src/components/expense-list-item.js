import React from 'react'
import {NavLink} from 'react-router-dom'
import moment from 'moment'

const ExpenseListItem = ({id, description, amount, createdAt } = {}) => (
    <div>
        <h3><NavLink to={`/edit/${id}`}>{description}</NavLink></h3>
        <p>{amount} - {moment(createdAt).format('MMM Do YYYY')}</p>
        
    </div>
)

export default (ExpenseListItem)