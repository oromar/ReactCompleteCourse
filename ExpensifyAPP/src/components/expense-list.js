import React from 'react'
import {connect} from 'react-redux'
import ExpenseListItem from './expense-list-item'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {
            props.expenses.length === 0 ? (<p>No Items Found</p>)
            : 
            (props.expenses.map((e) => <ExpenseListItem key={e.id} {...e} />))
        }
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList)
