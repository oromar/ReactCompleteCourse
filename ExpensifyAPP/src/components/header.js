import React from 'react'
import {NavLink} from 'react-router-dom'

const Header = () => 
    <div>
        <header>
            <h1>Expensify APP </h1>
        </header>
        <ul>
            <li><NavLink to='/' exact={true} activeClassName='is-active'>Dashboard</NavLink></li>
            <li><NavLink to='/create' activeClassName='is-active'>Create</NavLink></li>
            <li><NavLink to='/help' activeClassName='is-active'>Help</NavLink></li>
        </ul>
    </div>

export default Header
