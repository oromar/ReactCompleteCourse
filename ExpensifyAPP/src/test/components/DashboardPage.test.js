import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {ExpanseDashboardPage} from '../../components/dashboard'

test('should render dashboard correctly', () => {
    const wrapper = shallow(<ExpanseDashboardPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
})