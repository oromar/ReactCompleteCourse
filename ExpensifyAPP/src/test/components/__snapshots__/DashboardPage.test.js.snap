// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render dashboard correctly 1`] = `
<div>
  <h1>
    Dashboard
  </h1>
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
</div>
`;
